SwiftUIDemoAppUITests/Tests/DeeplinkTests.swift

```diff
@@ -0,0 +1,147 @@
+//  SwiftUIDemoAppUITests
+//
+//  Created by Ilias Pavlidakis on 25/9/23.
+//
+
+import XCTest
+
+final class DeeplinkTests: StreamTestCase {
+
+    static override func setUp() {
+        super.setUp()
+
+        // We are launching and terminating the app to ensure the executable
+        // has been installed.
+        app.launch()
+        app.terminate()
+    }
+
+    override func setUpWithError() throws {
+        launchApp = false
+        try super.setUpWithError()
+    }
+
+    private enum MockDeeplink {
+        static let production: URL = .init(string: "https://getstream.io/video/demos?id=test-call")!
+        static let customScheme: URL = .init(string: "streamvideo://video/demos?id=test-call")!
+    }
+
+    func test_associationFile_validationWasSuccessful() throws {
+        let contentData = try Data(contentsOf: .init(string: "https://getstream.io/.well-known/apple-app-site-association")!)
+        let content = try XCTUnwrap(String(data: contentData, encoding: .utf8))
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+            .replacingOccurrences(of: "\n", with: "")
+            .replacingOccurrences(of: " ", with: "")
+
+        XCTAssertEqual(content, """
+        {"applinks":{"apps":[],"details":[{"appID":"EHV7XZLAHA.io.getstream.iOS.VideoDemoApp","paths":["/video/demos/*","/video/demos"]}]}}
+        """)
+    }
+
+    func test_universalLink_production_joinsExpectedCall() {
+        WHEN("") {
+            Safari.openUniversalLinkFromSmartBanner(MockDeeplink.production)
+        }
+        THEN("user joins the the specified call") {
+            userRobot
+                .assertCallControls()
+                .assertParticipantsAreVisible(count: 1)
+        }
+    }
+
+    func test_customSchemeURL_joinsExpectedCall() {
+        WHEN("User opens a URL that contains a custom scheme") {
+            Safari.openApp(MockDeeplink.customScheme)
+        }
+        THEN("user joins the the specified call") {
+            userRobot
+                .assertCallControls()
+                .assertParticipantsAreVisible(count: 1)
+        }
+    }
+}
+
+extension XCUIDevice {
+
+    fileprivate func open(_ url: URL) {
+        DispatchQueue.main.async {
+            // Need to async this because XCUIDevice.shared.system.open
+            // synchronously waits for a button to be pressed the first time.
+            self.system.open(url)
+        }
+    }
+}
+
+enum Safari {
+
+    private static let safari = XCUIApplication(bundleIdentifier: "com.apple.mobilesafari")
+
+    fileprivate static func go(to url: URL) {
+        safari.textFields["Address"].tap()
+        safari.typeText(url.absoluteString)
+        let goButton = safari.buttons["Go"]
+        if goButton.waitForExistence(timeout: 5) {
+            safari.tapKeyboardKey("Go")
+        }
+    }
+
+    fileprivate static func open(_ url: URL) {
+        safari.launch()
+
+        _ = safari.wait(for: .runningForeground, timeout: 5)
```

Comment:
Does this mean that every test will have to wait for 5 sec? Any chance we could wait for some element to appear instead of sleeping for 5 sec?